src/model/attendance-board.test.ts

```diff
@@ -0,0 +1,38 @@
+import { ID } from "./id";
+import { Subject } from "./subject";
+
+export type AttendanceBoard = {
+    readonly id: ID<AttendanceBoard>;
+    readonly subject: ID<Subject>;
+    readonly startFrom: Date;
+    readonly secondsFromStartToBeLate: number;
+    readonly secondsFromBeLateToEnd: number;
+};
+
+export type AttendanceState = 
+    | "TOO_EARLY"
+    | "ATTENDED"
+    | "BE_LATE"
+    | "CLOSED"
+;
+
+export function determineState(board: AttendanceBoard, toAttendAt: Date): AttendanceState {
+       
+    const currentMs = Date.now(); // 現在の日時を取得
+    const beLateMs = toAttendAt.getTime() + board.secondsFromStartToBeLate * 1000;;//出席扱いの時間
+    const endMs = toAttendAt.getTime() + board.secondsFromStartToBeLate * 1000 + board.secondsFromBeLateToEnd * 1000;//遅刻扱いの時間
+
+if (currentMs < toAttendAt.getTime()) {
+    return "TOO_EARLY";
+} 
+if (currentMs < beLateMs){
+    return "ATTENDED";
+}
+if (currentMs < endMs){
+    return "BE_LATE";
+}
+if(endMs < currentMs){
+    return "CLOSED";
+}
```

Comment:
```suggestion
    return "CLOSED";
```
これだと `endMs` と `currentMs` が同じ場合の条件が抜けています. そもそもここに到達した時点で残る選択肢は `"CLOSED"` しかないので, `if` は不要です.